### PR TITLE
study youtube and complete in JS

### PR DESCRIPTION
**Pull request by cathy651**
https://github.com/pomo651/algo/issues/2

### CODE REVIEW

File: Sorting/JS/JS_basic_flexible.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
+
+flexible & scalable rather than hard coding
+尽量让代码可以复用，不受个别例子影响
```
Comment (pomo651) [inline Sorting/JS/JS_basic_flexible.js]: delete empty lines. 
add comment in front of line 3 and 4: // xxxx

File: Sorting/JS/JS_basic_flexible.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+var a = ['Wind', 'Water', 'Fire'];  如何打印出 'Wind, Water, Fire' ?
```
Comment (pomo651) [inline Sorting/JS/JS_basic_flexible.js]: 1. JS 标准: String 统一用 双引号  "wind"
2.comment 用 // 如何打印...

File: Sorting/JS/JS_basic_flexible.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+var a = ['Wind', 'Water', 'Fire'];
```
Comment (pomo651) [inline Sorting/JS/JS_basic_flexible.js]: 同上  "Wind"

File: Sorting/JS/JS_basic_flexible.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+const delimiter = ‘, ';
```
Comment (pomo651) [inline Sorting/JS/JS_basic_flexible.js]: ‘, ' -> doesn't make sense. use  ", "

File: Sorting/JS/JS_basic_flexible.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+console.log(a.join(delimiter));  
+
+
+join 的基本用法
+The join() method creates and returns a new string. If an element is undefined, null or an empty array [], it is converted to an empty string.
```
Comment (pomo651) [inline Sorting/JS/JS_basic_flexible.js]: use //

File: Sorting/JS/JS_basic_flexible.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+const elements = ['Fire', 'Air', 'Water'];
```
Comment (pomo651) [inline Sorting/JS/JS_basic_flexible.js]: 同上

File: Sorting/JS/JS_basic_flexible.js (hunk 1 above)
Comment (cathy651) [reply Sorting/JS/JS_basic_flexible.js]: done

File: Sorting/JS/JS_basic_flexible.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+// flexible & scalable rather than hard coding
+// 尽量让代码可以复用，不受个别例子影响
+
+// 题目var a = ["Wind", "Water", "Fire"];  如何打印出 "Wind, Water, Fire" ?
+
+var a = ["Wind", "Water", "Fire"];
```
Comment (pomo651) [inline Sorting/JS/JS_basic_flexible.js]: const replace var...

File: Sorting/JS/JS_basic_flexible.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+console.log(a.join(delimiter));
+
+// join 的基本用法
+// The join() method creates and returns a new string. If an element is undefined, null or an empty array [], it is converted to an empty string.
+
+const elements = ["Fire", "Air", "Water"];
+console.log(elements.join());
+// expected output: ‘Fire,Air,Water’
+
+var a = ["Wind", "Water", "Fire"];
```
Comment (pomo651) [inline Sorting/JS/JS_basic_flexible.js]: const replace var

**Review by pomo651 — APPROVED** [review]

**Pull request merged**